src/datachain/lib/dc.py

```diff
@@ -1624,6 +1625,208 @@ def subtract(  # type: ignore[override]
+        right_on = other.signals_schema.resolve(*right_on).db_signals()  # type: ignore[assignment]
+        if compare:
+            right_compare = right_compare or compare
+            compare = self.signals_schema.resolve(*compare).db_signals()  # type: ignore[assignment]
+            right_compare = other.signals_schema.resolve(*right_compare).db_signals()  # type: ignore[assignment]
+        else:
+            compare = [c for c in cols if c in right_cols]  # type: ignore[misc, assignment]
+            right_compare = compare
+
+        diff_cond = []
+
+        if added:
+            added_cond = sqlalchemy.and_(
+                *[
+                    C(c) == None  # noqa: E711
+                    for c in [f"{_rprefix(c, rc)}{rc}" for c, rc in zip(on, right_on)]
+                ]
+            )
+            diff_cond.append((added_cond, "A"))
+        if modified:
+            modified_cond = sqlalchemy.or_(
+                *[
+                    C(c) != C(f"{_rprefix(c, rc)}{rc}")
+                    for c, rc in zip(compare, right_compare)
+                ]
+            )
+            diff_cond.append((modified_cond, "M"))
+        if unchanged:
+            unchanged_cond = sqlalchemy.and_(
+                *[
+                    C(c) == C(f"{_rprefix(c, rc)}{rc}")
+                    for c, rc in zip(compare, right_compare)
+                ]
+            )
+            diff_cond.append((unchanged_cond, "U"))
+
+        diff = case(*diff_cond, else_=None).label(status_col)
+
+        left_right_merge = self.merge(
+            other, on=on, right_on=right_on, inner=False, rname=rname
+        )._query.select(
+            *([C(c) for c in on] + [C(c) for c in cols if c not in on] + [diff])
+        )
+
+        right_left_merge = (
+            other.merge(self, on=right_on, right_on=on, inner=False, rname=rname)
+            ._query.select(
+                *(
+                    [
+                        C(c) if c == rc else literal(None).label(c)
+                        for c, rc in zip(on, right_on)
+                    ]
+                    + [
+                        C(c) if c in right_cols else literal(None).label(c)  # type: ignore[arg-type]
+                        for c in cols
+                        if c not in on
+                    ]
+                    + [literal("D").label(status_col)]
+                )
+            )
+            .filter(
+                sqlalchemy.and_(
+                    *[
+                        C(f"{_rprefix(c, rc)}{c}") == None  # noqa: E711
+                        for c, rc in zip(on, right_on)
+                    ]
+                )
+            )
+        )
+
+        if not deleted:
+            res = left_right_merge
+        elif deleted and not any([added, modified, unchanged]):
+            res = right_left_merge
+        else:
+            res = left_right_merge.union(right_left_merge)
```

Comment:
please double check that it does not have duplicates after the joins 😅

Reply:
It shouldn't have any duplicated.